docs/index.rst

```diff
@@ -1,49 +1,138 @@
 
-This the landing page for documentation relevant to developers of the Open edX platform.  For other documentation
-resources targeted for edX learners, educators, researchers, and Open edX operators, see `docs.edx.org <docs.edx.org>`_.
+####################################
+Open edX Developer Documentation Hub
+####################################
+
+Welcome new and returning Open edX developers!
+
+This is the central repository for documentation that will help you work with Open edX code.
+
+
+New to Open edX development? Start here
+=======================================
+
+This quick overview is intended for developers new to the Open edX platform, junior and senior alike.
+
+After you've reviewed these topics you'll be ready for more detail on our architecture, or you can prepare for active development with the topics in "Information for all Open edX developers."
+
+* :doc:`new_to_edx/mission`
+* :doc:`Our code of conduct <universal_stub>`
+* `A 15-minute introduction to Open edX systems <https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/architecture.html>`_
+* :doc:`More about each system <new_to_edx/what_our_systems_do>`
+* :doc:`What you as a developer can do with the Open edX platform <new_to_edx/what_you_can_do>`
+* :doc:`all_development/roadmap/index`
 
 .. toctree::
-    :hidden:
+   :hidden:
+   :caption: New to Open edX development?
+
+   new_to_edx/mission
+   Our code of conduct <universal_stub>
+   A 15-minute introduction to Open edX systems <https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/architecture.html>
+   More about each system <new_to_edx/what_our_systems_do>
+   What you as a developer can do with the Open edX platform <new_to_edx/what_you_can_do>
+   all_development/roadmap/index
 
-    Open edX Proposals <https://open-edx-proposals.readthedocs.io/en/latest/>
-    developers_guide/index
-    named_releases
-    Open edX Extensions and APIs <https://open.edx.org/extending-edx>
-    (Legacy) Open edX Developer's Guide <https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/>
 
+Information for all Open edX developers
+=======================================
 
-Architecture and Best Practices
--------------------------------
+You will likely use some or all of the information in this section as you progress
+in your Open edX development work. Take a look through the topics here before you begin
+writing code.
+
+
+.. From the outline: these topics appear to be of common interest. Are they? i18n, XSS, MFEs (Micro-frontends), IDAs (Independently Deployed Applications, a.k.a. Microservices), Django Configuration/Toggles, Named Releases]
 
 .. list-table::
+   :widths: auto
 
-   * - Open edX Technology Radar (coming...)
-     - A bird's eye view of the decisions, tools and technologies that have been adopted or rejected for Open edX
-       development.
+   * - :doc:`all_development/roadmap/index`
```

Comment:
As I've returned to this, I have a couple of new comments related to `New to Open edX development? Start here` and `Information for all Open edX developers`.  Here is some food for thought:
* In the text (or in comments), it would be great to clarify the difference between these two, or I think it will get messy fast.  The clarification would both help a reader find what they are looking for, and a new author know where to add new docs. One possibility might be clarifying that the "New to Open edX" section should contain overviews and orientation information useful in the first 1 week (1-2 weeks?).
* Rather than pointing back to just one link (dev setup), maybe we should replace this with a link to the entire "New to Open edX" section with a description of what you'll find there.
* In actual docs, I could imagine deep dive docs pointing back to an overview of the same topic, or overviews pointing to deep dives, noting that the deep dive might not be needed immediately.

Reply:
@robrap Going forward let's put content-related comments on the PRs specific to that content. I think it will be easy for items to get lost in the shuffle otherwise, especially since this PR is now closed.